.changelog/12324.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+fix data race in TestAgentConfigWatcherSidecarProxy
```

Comment:
Do we usually include changelog entries for test fixes? I thought these were only for things that had user impact.

Reply:
There's an actual production race in the data that the test just happens to surface so i figured it was worth calling out. This is not a race in a test per se.

Reply:
Ah, I did not realize that from the changelog. Maybe something like:

> local: fixes a data race in anti-entropy sync that could cause the wrong tags to be applied to a service when EnableTagOverride is used

No idea if that is accurate, but something along those lines would probably help future readers understand the bug

Reply:
updated